### ISSUE TITLE
run wf-artic with conda --> unable to create Conda environment

### ISSUE THREAD

**Issue by MantaRay87** [epi2me-labs/wf-artic#36]
Hey!

I want to run the artic wf via conda and ran into multiple problems. Some I could solve myself but still want to bring them to your attention and one I could not and hope for help:

1. When I downloaded the latest version it had really weird user rights, in the sense that me as a user was not even able to execute it. I changed this via `chmod ` command. I also had to do it in the subfolders.

2. when I ran:` nextflow run ./main.nf -profile conda --fastq /ngs_rohdaten/fastq_pass/ --sample_sheet  /ngs_rohdaten/fastq_pass/sample_sheet.csv --scheme_version V4.1 --update_data`

I got the following message: 

```bash
WARN: Access to undefined parameter `samples` -- Initialise it to a default value eg. `params.samples = some_value`
Unparseable date: "2021-12-09T18_09_18Z"
```

Not sure why this nextclade_data_tag is unparsable but the earlier once (e.g `2021-06-25T00_00_00Z`) seem to work fine. I solved this problem by manually adding the flag ` --nextclade_data_tag 2022-01-05T19_54_31` because I thought using the most recent is the best. Is this correct or do you have a better idea? 

3. what I could not solve:

When now running the wf (`nextflow run ./main.nf -profile conda --fastq /ngs_rohdaten/fastq_pass/ --sample_sheet  /ngs_rohdaten/fastq_pass/sample_sheet.csv --scheme_version V4.1 --nextclade_data_tag 2022-01-05T19_54_31Z`) I get the following error:

```bash
WARN: Access to undefined parameter `samples` -- Initialise it to a default value eg. `params.samples = some_value`
Checking fastq input.
Barcoded directories detected.
Checking sample sheet.
executor >  local (93)
[76/02902b] process > checkSampleSheet         [100%] 1 of 1 ✔
[22/dbed04] process > pipeline:getVersions     [100%] 1 of 1 ✔
[40/98a28f] process > pipeline:getParams       [100%] 1 of 1 ✔
[04/d42fee] process > pipeline:copySchemeDir   [100%] 1 of 1 ✔
[bc/44301b] process > pipeline:preArticQC (12) [100%] 13 of 13 ✔
[80/0549f6] process > pipeline:runArtic (7)    [100%] 13 of 13 ✔
[c8/5c9338] process > pipeline:allConsensus    [100%] 1 of 1 ✔
[90/255c6c] process > pipeline:allVariants     [100%] 1 of 1 ✔
[03/b5bd74] process > pipeline:prep_nextclade  [100%] 1 of 1 ✔
[cc/33a289] process > pipeline:nextclade       [100%] 1 of 1 ✔
[-        ] process > pipeline:pangolin        -
[bc/86cdfe] process > pipeline:telemetry       [100%] 1 of 1 ✔
[-        ] process > pipeline:report          -
[1e/f36b17] process > output (58)              [100%] 58 of 58
```
  
```bash
Error executing process > 'pipeline:pangolin'

Caused by:
  Failed to create Conda environment
  command: conda create --mkdir --yes --quiet --prefix /data/SARS-CoV-2/ONT_artic_wf_aktuell/wf-artic-master/work/conda/env-d26fa193d248355ac3ce39abc12fab9e bioconda::pangolin=3.1.17 conda-forge::git
  status : 143
  message:
```
but the error message is empty. Could somebody help me? That would be really great!

**Comment by MantaRay87** [comment]
I played around and also ran 

`nextflow run ./main.nf -profile conda --fastq /srv/ngs_rohdaten/fastq_pass/ --sample_sheet /srv/ngs_rohdaten/fastq_pass/sample_sheet.csv --scheme_version V4.1 --nextclade_data_tag 2022-01-05T19_54_31Z --update_data`

Here I got two different but similar errors:

```
Checking fastq input.
Barcoded directories detected.
Checking sample sheet.
executor >  local (27)
executor >  local (28)
executor >  local (29)
executor >  local (30)
executor >  local (30)
executor >  local (31)
executor >  local (31)
[f9/d97547] process > checkSampleSheet         [100%] 1 of 1 ✔
executor >  local (31)
[f9/d97547] process > checkSampleSheet         [100%] 1 of 1 ✔
executor >  local (34)
[f9/d97547] process > checkSampleSheet         [100%] 1 of 1 ✔
executor >  local (34)
[f9/d97547] process > checkSampleSheet         [100%] 1 of 1 ✔
executor >  local (36)
[f9/d97547] process > checkSampleSheet         [100%] 1 of 1 ✔
[cd/c519f3] process > pipeline:getVersions     [100%] 1 of 1 ✔
[41/017220] process > pipeline:getParams       [100%] 1 of 1 ✔
executor >  local (36)
[f9/d97547] process > checkSampleSheet         [100%] 1 of 1 ✔
[cd/c519f3] process > pipeline:getVersions     [100%] 1 of 1 ✔
[41/017220] process > pipeline:getParams       [100%] 1 of 1 ✔
executor >  local (36)
[f9/d97547] process > checkSampleSheet         [100%] 1 of 1 ✔
[cd/c519f3] process > pipeline:getVersions     [100%] 1 of 1 ✔
[41/017220] process > pipeline:getParams       [100%] 1 of 1 ✔
[8a/192826] process > pipeline:copySchemeDir   [100%] 1 of 1 ✔
[96/660cbb] process > pipeline:preArticQC (10) [100%] 13 of 13 ✔
[7c/3fcfd2] process > pipeline:runArtic (7)    [100%] 13 of 13 ✔
[f4/b39a18] process > pipeline:allConsensus    [100%] 1 of 1 ✔
[ad/845347] process > pipeline:allVariants     [100%] 1 of 1 ✔
[82/05b0fd] process > pipeline:prep_nextclade  [100%] 1 of 1 ✔
[24/cc24e1] process > pipeline:nextclade       [  0%] 0 of 1
[-        ] process > pipeline:pangolin        -
[36/c786cd] process > pipeline:telemetry       [  0%] 0 of 1
[-        ] process > pipeline:report          -
[65/6f2046] process > output (1)               [100%] 1 of 1
/srv/data/SARS-CoV-2/ONT_artic_wf_aktuell/wf-artic-master/work/f0/2e6df243e2e0fc7abd4ac4a8fadc7a/NTC.consensus.fasta
/srv/data/SARS-CoV-2/ONT_artic_wf_aktuell/wf-artic-master/work/7c/3fcfd23165beb122754ca46bf5eeca/5307022.consensus.fasta
Creating Conda env: bioconda::pangolin=3.1.17 conda-forge::git [cache /srv/data/SARS-CoV-2/ONT_artic_wf_aktuell/wf-artic-master/work/conda/env-d26fa193d248355ac3ce39abc12fab9e]
Error executing process > 'pipeline:nextclade'

Caused by:
  Process `pipeline:nextclade` terminated with an error exit status (1)

Command executed:

  if [ "true" == "true" ]
  then
    nextclade dataset get --name 'sars-cov-2' --output-dir 'data/sars-cov-2' --tag 2022-01-05T19_54_31Z
  fi

  nextclade --version | sed 's/^/nextclade,/' > nextclade.version
  echo "nextclade_data_tag,`cat data/sars-cov-2/tag.json  | grep -Po '"tag": *\K"[^"]*"' | sed 's/\"//g'`" >> nextclade.version

  nextclade run         --input-fasta consensus.fasta         --reference data/sars-cov-2/reference.fasta         --input-pcr-primers primers.csv         --input-tree data/sars-cov-2/tree.json         --input-qc-config data/sars-cov-2/qc.js      executor >  local (36)
[f9/d97547] process > checkSampleSheet         [100%] 1 of 1 ✔
[cd/c519f3] process > pipeline:getVersions     [100%] 1 of 1 ✔
[41/017220] process > pipeline:getParams       [100%] 1 of 1 ✔
[8a/192826] process > pipeline:copySchemeDir   [100%] 1 of 1 ✔
[96/660cbb] process > pipeline:preArticQC (10) [100%] 13 of 13 ✔
[7c/3fcfd2] process > pipeline:runArtic (7)    [100%] 13 of 13 ✔
[f4/b39a18] process > pipeline:allConsensus    [100%] 1 of 1 ✔
[ad/845347] process > pipeline:allVariants     [100%] 1 of 1 ✔
[82/05b0fd] process > pipeline:prep_nextclade  [100%] 1 of 1 ✔
[24/cc24e1] process > pipeline:nextclade       [100%] 1 of 1, failed: 1 ✘
[-        ] process > pipeline:pangolin        -
[-        ] process > pipeline:telemetry       -
[-        ] process > pipeline:report          -
[65/6f2046] process > output (1)               [100%] 1 of 1
/srv/data/SARS-CoV-2/ONT_artic_wf_aktuell/wf-artic-master/work/f0/2e6df243e2e0fc7abd4ac4a8fadc7a/NTC.consensus.fasta
/srv/data/SARS-CoV-2/ONT_artic_wf_aktuell/wf-artic-master/work/7c/3fcfd23165beb122754ca46bf5eeca/5307022.consensus.fasta
Creating Conda env: bioconda::pangolin=3.1.17 conda-forge::git [cache /srv/data/SARS-CoV-2/ONT_artic_wf_aktuell/wf-artic-master/work/conda/env-d26fa193d248355ac3ce39abc12fab9e]
Error executing process > 'pipeline:nextclade'

Caused by:
  Process `pipeline:nextclade` terminated with an error exit status (1)

Command executed:

  if [ "true" == "true" ]
  then
    nextclade dataset get --name 'sars-cov-2' --output-dir 'data/sars-cov-2' --tag 2022-01-05T19_54_31Z
  fi

  nextclade --version | sed 's/^/nextclade,/' > nextclade.version
  echo "nextclade_data_tag,`cat data/sars-cov-2/tag.json  | grep -Po '"tag": *\K"[^"]*"' | sed 's/\"//g'`" >> nextclade.version

  nextclade run         --input-fasta consensus.fasta         --reference data/sars-cov-2/reference.fasta         --input-pcr-primers primers.csv         --input-tree data/sars-cov-2/tree.json         --input-qc-config data/sars-cov-2/qc.json         --input-gene-map data/sars-cov-2/genemap.gff         --output-json nextclade.json         --jobs 1

Command exit status:
  1

Command output:
  (empty)

Command error:
  [ERROR] Nextclade: Dataset not found: name='sars-cov-2', tag='2022-01-05T19_54_31Z'

Work dir:
  /srv/data/SARS-CoV-2/ONT_artic_wf_aktuell/wf-artic-master/work/24/cc24e1822a3f89e815f026bdf7610b

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`
```

**Comment by mjfos2r** [comment]
I also got the same error WARN: Access to undefined parameter `samples` -- Initialise it to a default value eg. `params.samples = some_value` but I run nextflow using docker. Very curious as to why it suddenly broke. Perhaps the recent Nextclade/Nextstrain update broke some things?

**Comment by mattdmem** [comment]
Thanks for this. We'll take a look, it's odd because we run with conda as part of our automated testing and it passed.

Nextclade is defaulted to 1.8.0 in this version of `wf-artic`. Thanks to the excellent communication from the Nextclade developers we are aware of potential issues with their new version and will be looking to address very soon.

**Comment by mattdmem** [comment]
Samples error fixed as of v0.3.11 - please open another issue if your other issue still persists! Thanks.
